ncm-spma/src/main/pan/components/spma/functions.pan

```diff
@@ -447,3 +447,35 @@ function pkg_ronly = {
 
     pkg_repl(ARGV[0], version, arch, list('mustexist'));
 };
+
+@documentation{
+    desc =  Add config for default module stream for overriding module defualt streams
+    arg = name
+    arg = stream
+    arg = (optional) profiles (type list(), set these as default profile)
+    arg = (optional) modulemd_verison (moduelmd-defualts profile version,it's currently 1 on rhel8 release)
+}
+function module_add = {
```

Comment:
out of curiosity, why do you have a function for this? is it to protect overriding existing streams (cfr th error)?

Reply:
partly that, similar to pkg_add_repl wanted to do any checks in one place.   then in pan, we can just use module_add(..)
but if you have better suggestion, then I'm all for it.

Reply:
depends a bit hat you want to configure imho. you can also use structure template(s) or plain diretc pan like
```
prefix /software/modules;
'python36/stream' = '3.6';
...
```

all i am saying is that function doesn't do that much imho.

Reply:
So all depends on what we want here, should the last entry win? function checks for duplicate definition of a module and fails compile. AFIAK you can't activate/set default  of two streams of same module,  so we can either silently take the last entry or fail compile. Function is my way of providing this check and have it fail compile to alert user that two stream of same module cannot be  enabled.

taking your approach, but with a module that has different streams,  
```
prefix "/software/modules";
"postgresql/stream" = "11";
```
then somewhere else in template, (through another feature maybe), someone adds following, 
```
prefix "/software/modules";
"postgresql/stream" = "10";
```
should this fail compile?

setting through the function, it will fail compile. i.e 
```

"/software/modules" = module_add("postgresql", "11", list("server"));
....
"/software/modules" = module_add("postgresql", "10", list("server"));

```
but anyway maybe this is our use case and there are better ways to do this? So I'll be happy to remove the function if this isn't desirable upstream?

Reply:
@aka7 to be clear, your function is desirable ;) but the pan code over time has accumulated lots of code and variables that imho would be better hamdled with using path.

so just so you know, the function does not prevent anyone to override it later using the direct path.
there are several ways to deal with this problem, but all have their drawbacks.

if you want to hard protect a stream path from alteration, you can use `final` (but that also has drawbacks).
defaults are also nice to use.

we are used to having the `last one wins`, but you might have different experience

Reply:
@stdweird we are also used to having last one wins, and you make valid point about function doesn't prevent someone from setting using direct path.   I think this probably requires further discussion if we are to provide a solution for duplicate entries where last one should not win.  This is why I'm suggesting maybe I remove function from pr for now for further discussion, mainly  so I can get rest of code in. We are keen to get a new point release with this change, if possible. If the rest of the code makes sense.

Reason for the rush is that, redhat annoying made lot of modules default which come EOL next year, where as they have different stream with longer support, but are not set as the default. We like to make all long term support modules the default before we release RHEL8 into the wild.  if that makes sense.

Reply:
I had a offline conversation with @jrha  and we agreed to remove the function. This is now updated.